lib/internal/source_map.js

```diff
@@ -0,0 +1,19 @@
+'use strict';
+
+const _sourceMapCache = {};
+function maybeStoreSourceMap(filename, content) {
+  if (!process.env.NODE_V8_COVERAGE) return;
+  const match = content.match(/\/[*/]#\s+sourceMappingURL=(?<sourceMap>[^\s]+)/);
+  if (match) {
+    _sourceMapCache[filename] = match.groups.sourceMap;
+  }
+}
+
+function _getSourceMapCache() {
+  return _sourceMapCache;
+}
```

Comment:
If the object is going to be public, we may want to put a bit more mechanism in place here than just exposing a raw object. Should this object *be* openly modifiable by user code? Should there be an API around it? Should it be a `Map` instead? I think landing this as it is currently would be fine so long as the entire mechanism is marked experimental but before it becomes a supported API we should revisit this.

Reply:
:+1: went ahead used a Map, which seems to make more sense for a cache. The method `sourceMapCacheToObject` is meant ot be a private helper for serializing the object in the inspector.